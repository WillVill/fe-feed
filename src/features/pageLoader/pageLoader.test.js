import React from 'react';
import LoaderPage from './pageLoader.component.jsx';
// Importing shallow and toJson.
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

// Test Setup
const setup = propOverrides => {
	const props = Object.assign({
		text: "loading..."
	}, propOverrides);

	const wrapper = shallow(<LoaderPage {...props} />);

	return {
		props,
		wrapper
	};
};

describe('[Feature] pageLoader', () => {
	it('renders', () => {
		const { wrapper } = setup();
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('passes proper text prop', () => {
		const { wrapper } = setup();
		expect(wrapper.find('Loader').props().children).toEqual("loading...");
	});

	it('passes proper changed text prop', () => {
		const { wrapper } = setup({text: "Gnarley text"});
		expect(wrapper.find('Loader').props().children).toEqual("Gnarley text");
	});
});

