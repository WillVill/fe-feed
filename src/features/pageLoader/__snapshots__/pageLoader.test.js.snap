// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Feature] pageLoader renders 1`] = `
<div
  className="loader-wrapper"
>
  <Loader
    active={true}
    size="large"
  >
    loading...
  </Loader>
</div>
`;
